### PR TITLE
Yank 0.11.0: The YAML documentation cometh

### LINKED ISSUE #496: Add more complete documentation for every options/key that can be set in the `.yaml` files.

### PR DESCRIPTION
**Pull request by Lnaden**
- Massive overhaul to YAML documentation. All options now specified in the RST docs. 
- Updated guidelines for making releases and updates.
- Enough here to merit a X.Y+1.Z update. 

Fixes #496

### CODE REVIEW

**Comment by Lnaden** [comment]
For review, please pull down the PR/branch, `cd docs`, `make hrml` then drop `_build/index.rst` into a browser window to see the changes.

**Comment by andrrizzi** [comment]
Reviewing now!

File: docs/yamlpages/protocols.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+* :ref:`Hydration Free Energies Setup by YANK <yaml_systems_hydration>`
+
+  * ``{PhaseName1}`` -> ``solvent1``
+  * ``{PhaseName2}`` -> ``solvent2``
+
+* :ref:`Arbitrary Phase Free Energies Setup by User <yaml_systems_user_defined>`
+
+  * ``{PhaseName1}`` -> ``phase1``
+  * ``{PhaseName2}`` -> ``phase2``
```
Comment (andrrizzi) [inline docs/yamlpages/protocols.rst]: Ok, I did not document very well phase names, sorry. The phase names don't have to match exactly those names as long if they contains those keywords. For example, 'my_complex_phase' and 'blablasolventbla` are accepted and they match a binding free energy calculation. It is also possible to specify any name if you use the `!Ordered` keyword (I've enforced this to make it clear to the user that the order of the phases determines the sign of the result). For example
```yaml
absolute-binding: !Ordered
  plus_sign_phase:
    alchemical_path: ...
  minus_sign_phase:
    alchemical_path: ...
```

File: docs/yamlpages/solvents.rst [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+-----------
+.. code-block:: yaml
+
+   solvents:
+     {UserDefinedSolvent}:
+       constraints: Hbonds
+
+.. todo:: Figure out which ``constraints`` option overwrites the other.
+
+Specify constraints over the entire system. This option is redundant with the general options for :ref:`constraints <yaml_options_constraints>` and should almost never be set here unless you want to specify two solvents with different constraints.
```
Comment (andrrizzi) [inline docs/yamlpages/solvents.rst]: Good point! We could think of adding an exception for this option in the schema validator.

File: docs/yamlpages/systems.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
+.. _yaml_systems_head:
+
+Systems Header for YAML Files
+*****************************
+
+The ``systems`` header is where we combine :doc:`molecules <molecules>` and :doc:`solvents <solvents>` together with molecular forcefields and general options to create the environemnt we want to simulation.
```
Comment (andrrizzi) [inline docs/yamlpages/systems.rst]: Typo: environemnt

File: docs/yamlpages/systems.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+       receptor: {UserDefinedMolecule}
+       ligand: {UserDefinedMolecule}
+       solvent: {UserDefinedSolvent}
+       pack: yes
+       leap:
+           parameters: [leaprc.ff14SB, leaprc.gaff]
+
+This is the mandatory structure for leting YANK handle the creation and setup of a Ligand/Receptor system to do binding free energies.
+
+* ``recptor``: Tells YANK which ``{UserDefinedMolecule}`` to load in as the receptor.
```
Comment (andrrizzi) [inline docs/yamlpages/systems.rst]: Typo: recptor

**Comment by jchodera** [comment]
Do you guys need me to help with the update to the conda recipe at some point? Or is that documented now so that you can tackle it yourselves?

**Comment by Lnaden** [comment]
I should be able to take care of it

**Comment by Lnaden** [comment]
So the builds appear to be having issues in that `numpy` is having issues loading for the `conda-build` call. Any idea what is going on there? 

https://travis-ci.org/choderalab/yank/jobs/164926897#L1677-L1698

**Comment by jchodera** [comment]
Not sure what is causing that, but it seems to be related to the mix_repkicas compiled code. I forget what scheme we are using to compile mix_replicas. I wonder if an upstream package was just updated.

Did the overnight travis build of master also fail?

**Comment by Lnaden** [comment]
The nightly appears to be fine. I don't think it necessarily has to do with the _mix_replicas.so since the entire traceback has to do with `conda-build`.

**Comment by andrrizzi** [comment]
The nightly build seems to have the same error: https://travis-ci.org/choderalab/yank/jobs/164793722#L1691

**Comment by Lnaden** [comment]
I'll tackle it this afternoon, and figure out what changed

**Comment by Lnaden** [comment]
Okay, tests now passing, I'll tag it and ship it!

**Comment by jchodera** [comment]
cc: https://github.com/msmbuilder/msmbuilder/pull/915#issuecomment-251507007

**Pull request merged**